### PR TITLE
refactor: use helpers for command-line parsing in renderer/init.js

### PR DESCRIPTION
**Pull request by miniak**
#### Description of Change
Makes the command-line parsing more robust.

#### Checklist
- [x] PR description included and stakeholders cc'd
- [x] `npm test` passes
- [x] PR title follows semantic [commit guidelines](https://github.com/electron/electron/blob/master/docs/development/pull-requests.md#commit-message-guidelines)

#### Release Notes
Notes: no-notes

### CODE REVIEW

File: lib/renderer/command-line.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
+'use strict'
+
+exports.hasSwitch = function (name) {
+  return process.argv.includes(`--${name}`)
```
Comment (nornagon) [inline lib/renderer/command-line.js]: This should be implemented as a call through to the C++ `CommandLine::HasSwitch`, which has more robust logic than this. Same for `getSwitchValue` below.

Comment (miniak) [reply lib/renderer/command-line.js]: done

**Comment by miniak** [comment]
@nornagon I've exposed `hasSwitch` and `GetSwitchValue` wrapping `base::CommandLine` from native code

File: atom/browser/web_contents_preferences.cc [hunk 2]
```diff
@@ -228,19 +228,17 @@ void WebContentsPreferences::AppendCommandLineSwitches(
...
-                                  enable_node_integration ? "true" : "false");
+  if (IsEnabled(options::kNodeIntegration))
+    command_line->AppendSwitch(switches::kNodeIntegration);
 
   // Whether to enable node integration in Worker.
   if (IsEnabled(options::kNodeIntegrationInWorker))
     command_line->AppendSwitch(switches::kNodeIntegrationInWorker);
 
   // Check if webview tag creation is enabled, default to nodeIntegration value.
   // TODO(kevinsawicki): Default to false in 2.0
```
Comment (deepak1556) [inline atom/browser/web_contents_preferences.cc]: Can you remove this TODO comment

File: atom/common/api/atom_api_command_line.cc [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+#include "atom/common/node_includes.h"
+
+namespace {
+
+bool HasSwitch(const base::CommandLine::StringType& name) {
+  return base::CommandLine::ForCurrentProcess()->HasSwitch(name);
+}
+
+base::CommandLine::StringType GetSwitchValue(
```
Comment (deepak1556) [inline atom/common/api/atom_api_command_line.cc]: `#include "atom/common/native_mate_converters/string16_converter.h"` for `StringType` on windows.

File: atom/browser/web_contents_preferences.cc [hunk 4]
```diff
@@ -228,19 +228,17 @@ void WebContentsPreferences::AppendCommandLineSwitches(
         ::switches::kEnableExperimentalWebPlatformFeatures);
 
   // Check if we have node integration specified.
-  bool enable_node_integration = IsEnabled(options::kNodeIntegration, true);
-  command_line->AppendSwitchASCII(switches::kNodeIntegration,
-                                  enable_node_integration ? "true" : "false");
+  if (IsEnabled(options::kNodeIntegration))
```
Comment (deepak1556) [inline atom/browser/web_contents_preferences.cc]: This makes default to `false`, can this be done without the changes from https://github.com/electron/electron/pull/16235 ?

Comment (miniak) [reply atom/browser/web_contents_preferences.cc]: it does not, the defaults are set here
```
  // Set WebPreferences defaults onto the JS object
  SetDefaultBoolIfUndefined(options::kPlugins, false);
  SetDefaultBoolIfUndefined(options::kExperimentalFeatures, false);
  bool node = SetDefaultBoolIfUndefined(options::kNodeIntegration, true);
  SetDefaultBoolIfUndefined(options::kNodeIntegrationInWorker, false);
  SetDefaultBoolIfUndefined(options::kWebviewTag, node);
  SetDefaultBoolIfUndefined(options::kSandbox, false);
  SetDefaultBoolIfUndefined(options::kNativeWindowOpen, false);
  SetDefaultBoolIfUndefined(options::kContextIsolation, false);
```

Comment (miniak) [reply atom/browser/web_contents_preferences.cc]: the default value is changed later in https://github.com/electron/electron/pull/16235

**Comment by miniak** [comment]
@robo all comments should be addressed

**Comment by miniak** [comment]
@deepak1556 all comments should be addressed

**Comment by jkleinsc** [comment]
Merging with failing arm test as it is broken in master

**Pull request merged**